### PR TITLE
docs: Add instructions for running for WASM

### PR DESCRIPTION
**Pull request by radical**

### CODE REVIEW

**Review by adamsitnik — APPROVED** [review]
LGTM, thank you very much @radical !

File: docs/benchmarking-workflow-dotnet-runtime.md [hunk 1]
```diff
@@ -139,6 +140,42 @@ cp artifacts\bin\runtime\net7.0-Windows_NT-Release-x64\Microsoft.Extensions.Cach
...
+1. build the dotnet/runtime repository in **Release**:
+
+```cmd
+/path/to/dotnet/runtime$ ./build.sh mono+libs -os browser -c Release
+```
+
+2. Prepare a sdk with `wasm-tools` workload installed using the built artifacts:
+
+```cmd
+/path/to/dotnet/runtime$ dotnet build -p:TargetOS=Browser -p:TargetArchitecture=wasm -c Release src/mono/wasm/Wasm.Build.Tests /t:InstallWorkloadUsingArtifacts
```
[the remark this answers is not in the document]
Comment (adamsitnik) [reply docs/benchmarking-workflow-dotnet-runtime.md:156]: @radical after I apply some changes locally and I rebuilt the repo, do I just need to run this command again?

[the remark this answers is not in the document]
Comment (radical) [reply docs/benchmarking-workflow-dotnet-runtime.md:156]: If you are making changes to the runtime, and any wasm build stuff then yeah.

**Comment by radical** [comment]
@adamsitnik there are markdown lint errors, unrelated to the change here. What should I do about those?

```
Error: docs/benchmarkdotnet.md:7:3 MD051/link-fragments Link fragments should be valid [Context: "[Main Concepts](#Main-Concepts)"]
[10](https://github.com/dotnet/performance/actions/runs/3110388058/jobs/5041518879#step:4:11)
Error: docs/benchmarkdotnet.md:8:3 MD051/link-fragments Link fragments should be valid [Context: "[Prerequisites](#Prerequisites)"]
[11](https://github.com/dotnet/performance/actions/runs/3110388058/jobs/5041518879#step:4:12)
Error: docs/benchmarkdotnet.md:9:3 MD051/link-fragments Link fragments should be valid [Context: "[Building the benchmarks](#Building-the-Benchmarks)"]
[12](https://github.com/dotnet/performance/actions/runs/3110388058/jobs/5041518879#step:4:13)
Error: docs/benchmarkdotnet.md:10:5 MD051/link-fragments Link fragments should be valid [Context: "[Using .NET Cli](#Build-using-.NET-Cli)"]
...
```

**Comment by DrewScoggins** [comment]
> @adamsitnik there are markdown lint errors, unrelated to the change here. What should I do about those?
> 
> ```
> Error: docs/benchmarkdotnet.md:7:3 MD051/link-fragments Link fragments should be valid [Context: "[Main Concepts](#Main-Concepts)"]
> [10](https://github.com/dotnet/performance/actions/runs/3110388058/jobs/5041518879#step:4:11)
> Error: docs/benchmarkdotnet.md:8:3 MD051/link-fragments Link fragments should be valid [Context: "[Prerequisites](#Prerequisites)"]
> [11](https://github.com/dotnet/performance/actions/runs/3110388058/jobs/5041518879#step:4:12)
> Error: docs/benchmarkdotnet.md:9:3 MD051/link-fragments Link fragments should be valid [Context: "[Building the benchmarks](#Building-the-Benchmarks)"]
> [12](https://github.com/dotnet/performance/actions/runs/3110388058/jobs/5041518879#step:4:13)
> Error: docs/benchmarkdotnet.md:10:5 MD051/link-fragments Link fragments should be valid [Context: "[Using .NET Cli](#Build-using-.NET-Cli)"]
> ...
> ```

Do we just need to make all of those links lowercase? If so could you just fix them up? If that doesn't work go ahead and just merge this, and we can fix it later.

**Comment by radical** [comment]
Markdown errors are fixed. But some unrelated failures are there now. Should I merge?

**Comment by radical** [comment]
Failures are unrelated, merging.

File: docs/benchmarking-workflow-dotnet-runtime.md [hunk 2]
```diff
@@ -139,6 +140,42 @@ cp artifacts\bin\runtime\net7.0-Windows_NT-Release-x64\Microsoft.Extensions.Cach
...
+    --bdn-arguments="--anyCategories Libraries Runtime --category-exclusion-filter NoInterpreter NoWASM NoMono --logBuildOutput --wasmDataDir </path/to/dotnet/runtime>/src/mono/wasm --filter <filter>"
+```
+
+#### Run the benchmarks with AOT:
+
+Essentially, add `--aotcompilermode wasm` to the `--bdn-arguments=".."`:
+
+```cmd
+/path/to/dotnet/performance$ python3 ./scripts/benchmarks_ci.py --csproj src/benchmarks/micro/MicroBenchmarks.csproj -f net7.0 --dotnet-path </path/to/dotnet/runtime/>artifacts/bin/dotnet-net7+latest --wasm --bdn-artifacts artifacts/BenchmarkDotNet.Artifacts
+    --bdn-arguments="--anyCategories Libraries Runtime --category-exclusion-filter NoInterpreter NoWASM NoMono --aotcompilermode wasm --logBuildOutput --buildTimeout 3600 --wasmDataDir </path/to/dotnet/runtime>/src/mono/wasm --filter <filter>"
```
[the remark this answers is not in the document]
Comment (adamsitnik) [reply docs/benchmarking-workflow-dotnet-runtime.md]: @radical when I run it it still fails with https://github.com/dotnet/performance/pull/2612#discussion_r976146339 and I need to comment it out every time. Do you know what can be causing this compile error? Is there any chance I am using outdated v8 or some other tool?

**Pull request merged**